safir/src/safir/testing/containers/_kafka.py

```diff
@@ -0,0 +1,341 @@
+"""Kafka test container with listeners for different security protocols.
+
+This test container is heavily inspired by the default Testcontainers_ `Kafka
+test container
+<https://github.com/testcontainers/testcontainers-python/blob/main/modules/kafka/testcontainers/kafka/__init__.py>`__,
+but adds additional listeners for different security protocols. The key
+changes are:
+
+* Create SSL, SASL_SSL, and SASL_PLAINTEXT listeners in addition to the
+  the PLAINTEXT listener.
+* Expose the SASL username and password with ``get_sasl_username`` and
+  ``get_sasl_password`` methods.
+* Provide a ``get_secret_file_contents`` method to retrieve the TLS
+  certificates created during startup.
+* Provide a ``reset`` method to delete all topics from the running Kafka
+  instance.
+* Always limit the broker to the first host.
+* Always use KRaft.
+"""
+
+import errno
+import os
+import sys
+import tarfile
+import time
+from importlib import resources
+from io import BytesIO
+from pathlib import Path
+from textwrap import dedent
+from typing import Any, Self
+
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.version import ComparableVersion
+from testcontainers.core.waiting_utils import wait_for_logs
+
+from ._constants import CONFLUENT_VERSION_TAG, KAFKA_REPOSITORY
+
+__all__ = ["FullKafkaContainer"]
+
+
+class FullKafkaContainer(DockerContainer):
+    """Kafka container.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+       from testcontainers.kafka import KafkaContainer
+
+       with KafkaContainer() as kafka:
+           connection = kafka.get_bootstrap_server()
+    """
+
+    _CLUSTER_ID = "MkU3OEVBNTcwNTJENDM2Qk"
```

Comment:
I don't _think_ there's anything magical about this value. I know it's used in the Kafka docker docs and a bunch of the random internet posts. We could probably change it if we want to, or say here that its arbitrary.